docs/other/configuration-with-osg-configure.md

```diff
@@ -0,0 +1,550 @@
+If something goes wrong, specify the `-d` flag to add more verbose output to `osg-configure.log`.
+
+The rest of this document will detail what to specify in the INI files.
+
+
+Conventions
+-----------
+
+In the tables below:
+
+-   Mandatory options for a section are given in **bold** type. Sometime the default value may be OK and no edit required, but the variable has to be in the file.
+-   Options that are not found in the default ini file are in *italics*.
+
+
+
+Syntax and layout
+=================
+
+The configuration files used by `osg-configure` are the one supported by Python's [SafeConfigParser](http://docs.python.org/library/configparser.html), similar in format to the [INI configuration file](http://en.wikipedia.org/wiki/INI_file) used by MS Windows:
+
+-   Config files are separated into sections, specified by a section name in square brackets (e.g. `[Section 1]`)
+-   Options should be set using `name = value` pairs
+-   Lines that begin with `;` or `#` are comments
+-   Long lines can be split up using continutations: each white space character can be preceded by a newline to fold/continue the field on a new line (same syntax as specified in [email RFC 822](http://tools.ietf.org/html/rfc822.html))
+-   Variable substitutions are supported -- [see below](#variable-substitution)
+
+`osg-configure` reads and uses all of the files in `/etc/osg/config.d` that have a ".ini" suffix. The files in this directory are ordered with a numeric prefix with higher numbers being applied later and thus having higher precedence (e.g. 00-foo.ini has a lower precedence than 99-local-site-settings.ini). Configuration sections and options can be specified multiple times in different files. E.g. a section called `[PBS]` can be given in `20-pbs.ini` as well as `99-local-site-settings.ini`.
+
+Each of the files are successively read and merged to create a final configuration that is then used to configure OSG software. Options and settings in files read later override the ones in previous files. This allows admins to create a file with local settings (e.g. `99-local-site-settings.ini`) that can be read last and which will be take precedence over the default settings in configuration files installed by various RPMs and which will not be overwritten if RPMs are updated.
+
+
+Variable substitution
+---------------------
+
+The osg-configure parser allows variables to be defined and used in the configuration file:
+any option set in a given section can be used as a variable in that section.  Assuming that you have set an option with the name `myoption` in the section, you can substitute the value of that option elsewhere in the section by referring to it as `%(myoption)s`.
+
+!!! note
+    The trailing `s` is required. Also, option names cannot have a variable subsitution in them.
+
+!!! warning
+    You will need to be careful when naming variables in order to avoid an
+    infinite loop when resolving the variable substitution.
```

Comment:
Could you offer specifics on exactly what to avoid doing?

Reply:
Actually, can I remove it? I doubt most people make variables any more... what are the chances?

Reply:
What's the failure mode of the infinite loop, a runaway osg-configure process? If they can just CTRL-C and revisit their config, I'm fine with getting rid of this

Reply:
A fairly descriptive Python error:
```
Unknown exception encountered while running: Value interpolation too deeply recursive:
        section: [RSV]
        option : ce_hosts
        rawval : foo %(htcondor_ce_hosts)s
```